runners/samza/src/main/java/org/apache/beam/runners/samza/SamzaPipelineRunner.java

```diff
@@ -36,8 +39,16 @@
 
   @Override
   public PortablePipelineResult run(final Pipeline pipeline, JobInfo jobInfo) {
+    // Expand any splittable DoFns within the graph to enable sizing and splitting of bundles.
```

Comment:
Considering the repetition between Flink, Spark and Samza, maybe worth moving the recurring sequence into "preparation" utility function?

Reply:
I was thinking about that but the preparation will likely diverge once more forms of splitting are added since the various graph expansions will change.